File: server/http/Model/SystemMail.cs
using StackExchange.Redis;

namespace Http.Model
{
    public class SystemMailKey : BaseModel, IRedisHashKey
    {
        public required uint Id { get; set; }
        public uint? GetHash() => 0;

        public RedisKey GetRedisKey() => "fb:cache:system_mail";

        public RedisValue GetRedisField() => Id.ToString();
    }

    public class SystemMail : SystemMailKey, IModel
    {
        public uint Sender { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Contents { get; set; } = string.Empty;
        public DateTime? ExpireDate { get; set; }
    }
}

File: server/game/src/system_mail.cpp
#include <fb/game/system_mail.h>
#include <fb/protocol/flatbuffer/protocol.h>

using namespace fb::game;

system_mail::system_mail(const fb::protocol::internal::SystemMail& dto) :
    id(dto.id),
    sender(dto.sender),
    title(dto.title),
    contents(dto.contents),
    created_date(fb::model::datetime(dto.created_date))
{
    if (dto.expire_date.has_value() && !dto.expire_date.value().empty())
        this->expire_date = fb::model::datetime(dto.expire_date.value());
    else
        this->expire_date = std::nullopt;
}
